keras/utils/hdf5_utils.py

```diff
@@ -0,0 +1,197 @@
+"""HDF5 related utilities."""
+import numpy as np
+import sys
+
+
+try:
+    import h5py
+    HDF5_OBJECT_HEADER_LIMIT = 64512
+except ImportError:
+    h5py = None
+
+
+if sys.version_info[0] == 3:
+    import pickle
+else:
+    import cPickle as pickle
+
+
+# FLAGS
+_is_boxed = '_is_dict_boxed'
+
+
+class H5Dict(object):
+    """ A dict-like wrapper around h5py groups (or dicts).
+
+    This allows us to have a single serialization logic
+    for both pickling and saving to disk.
+
+    Note: This is not intended to be a generic wrapper.
+    There are lot of edge cases which have been hardcoded,
+    and makes sense only in the context of model serialization/
+    deserialization.
+    """
+
+    def __init__(self, path, mode='a'):
+        if isinstance(path, h5py.Group):
+            self.data = path
+            self._is_file = False
+        elif type(path) is str:
```

Comment:
Do not use `type(...) is`, use `isinstance`. To test for string types, use `six`